python-package/xgboost/core.py

```diff
@@ -738,7 +738,7 @@ def feature_names(self, feature_names):
         if feature_names is not None:
             # validate feature name
             if not isinstance(feature_names, list):
-                feature_names = list(feature_names)
+                feature_names = [feature_names]
```

Comment:
I don't think this is the best way to solve the problem. I think a better way is to first check `isinstance(feature_names, STRING_TYPES)` and have a special handling logic for the single string case.